### ISSUE TITLE
make fails on Windows when trying to build from source

### ISSUE THREAD

**Issue by RicardoFernandez-UY** [gogatekeeper/gatekeeper#200]
Hi,
I'm trying to build from source on a Windows 10 machine.
I have go 1.19
I followed the steps given in : https://github.com/gogatekeeper/gatekeeper/blob/master/docs/building.md
And issued this command:  

`make && make test --no-silent`

It starts and displays:  

```
"--> Go Version"
go version go1.19 windows/amd64
```
Then it stays there for a long time showing nothing else, and when I hit enter it shows:

```
"--> Compiling the project"
go build -ldflags "-X main.gitsha=c8f31a2 -X main.compiled=The system cannot accept the date entered. Enter the new date: (mm-dd-yy) " -o bin/gatekeeper
# github.com/gogatekeeper/gatekeeper
usage: link [options] main.o
  -B note
        add an ELF NT_GNU_BUILD_ID note when using ELF
  -E entry
        set entry symbol name
  -H type
        set header type
  -I linker
        use linker as ELF dynamic linker
  -L directory
        add specified directory to library path
  -R quantum
        set address rounding quantum (default -1)
  -T address
        set text segment address (default -1)
  -V    print version and exit
  -X definition
        add string value definition of the form importpath.name=value
  -a    no-op (deprecated)
  -asan
        enable ASan interface
  -aslr
        enable ASLR for buildmode=c-shared on windows (default true)
  -benchmark string
        set to 'mem' or 'cpu' to enable phase benchmarking
  -benchmarkprofile base
        emit phase profiles to base_phase.{cpu,mem}prof
  -buildid id
        record id as Go toolchain build id
  -buildmode mode
        set build mode
  -c    dump call graph
  -compressdwarf
        compress DWARF if possible (default true)
  -cpuprofile file
        write cpu profile to file
  -d    disable dynamic executable
  -debugnosplit
        dump nosplit call graph
  -debugtextsize int
        debug text section max size
  -debugtramp int
        debug trampolines
  -dumpdep
        dump symbol dependency graph
  -extar string
        archive program for buildmode=c-archive
  -extld linker
        use linker when linking in external mode
  -extldflags flags
        pass flags to external linker
  -f    ignore version mismatch
  -g    disable go package data checks
  -h    halt on error
  -importcfg file
        read import configuration from file
  -installsuffix suffix
        set package directory suffix
  -k symbol
        set field tracking symbol
  -libgcc string
        compiler support lib for internal linking; use "none" to disable
  -linkmode mode
        set link mode
  -linkshared
        link against installed Go shared libraries
  -memprofile file
        write memory profile to file
  -memprofilerate rate
        set runtime.MemProfileRate to rate
  -msan
        enable MSan interface
  -n    dump symbol table
  -o file
        write output to file
  -pluginpath string
        full path name for plugin
  -r path
        set the ELF dynamic linker search path to dir1:dir2:...
  -race
        enable race detector
  -s    disable symbol table
  -strictdups int
        sanity check duplicate symbol contents during object file reading (1=warn 2=err).
  -tmpdir directory
        use directory for temporary files
  -v    print link trace
  -w    disable DWARF generation
make: *** [Makefile:28: build] Error 2
```

What could be the problem? Your help will be much appreciated!

Thank you!!!

**Comment by p53** [comment]
please use linux for this software, not sure what you are using on Win, probably cygwin, problem will be output of this command ``date '+%s'``, you can use WSL for compiling

**Comment by RicardoFernandez-UY** [comment]
Understood. Thank you!! 

I usually do my testing in Windows/Docker and then deploy in Linux. 
I first tried the docker example here https://github.com/gogatekeeper/gatekeeper, but it didn't work. 
Maybe because I have Keycloak also running in Docker, and I have one of those docker networking problems, or I'm missing something else on the gatekeeper side. I think that on Keycloak side things are ok because I've tested them with Postman and they returned values as expected.

I executed this command:

`docker run -it --rm quay.io/gogatekeeper/gatekeeper:1.6.0 --listen 127.0.0.1:28090 --upstream-url http://127.0.0.1:8000/admin --discovery-url http://127.0.0.1:28081/auth/realms/gatekeeper_realm --client-id gatekeeperclient`

I got this error:

`[error] failed to retrieve the provider configuration from discovery url: Get "http://127.0.0.1:28081/auth/realms/gatekeeper_realm/.well-known/openid-configuration": dial tcp 127.0.0.1:28081: connect: connection refused`

 Now I followed your advice for the "build from source" and did it in WSL and it worked fine.
Thank you very much!
